### DISCUSSION TITLE
`matchDepTypes` does not add label

### DISCUSSION THREAD

**Discussion by p3k** [renovatebot/renovate#27118] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.168.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### What is your question?

I am trying to add a label to merge requests for packages defined in the `dependencies` section of package.json:

```json
{
  "packageRules": [
    {
      "matchDepTypes": ["dependencies"],
      "addLabels": ["Runtime"]
    }
  ]
}
```

– but there is no label added to any package!

E.g. in the following example of package.json (excerpt) that should be `foo` and `bar`, but not `foobar`:

```json
{
  "dependencies": {
    "foo": "1.2.3",
    "bar": "0.0.1"
  },
  "devDependencies": {
    "foobar": "9.8.7"
  }
}
```

What am I doing wrong?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by mikaello** [comment]
`matchDepTypes` is valid only within the `packageRules` object (see [documentation](https://docs.renovatebot.com/configuration-options/#matchdeptypes)), try something like this instead:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "packageRules": [
    {
      "matchDepTypes": ["dependencies"],
      "addLabels": ["Runtime"]
    }
  ]
}
```

**Reply by p3k** [reply]
Yeah, this is just an excerpt, the full definition is in a package rule.

**Reply by p3k** [reply]
updated the question for clarity.

**Reply by mikaello** [reply]
Then I don't get it, I have copy-pasted the exact code in your question (just versions adjusted to actually get some updates), and it works like charm, labels are added: https://github.com/mikaello/renovate-matchdeptype-repro/pulls

![image](https://github.com/renovatebot/renovate/assets/2505178/8c80ff50-244b-466f-9a40-0266ad5422de)


(the default hourly limit is 2, that's why you don't see anything for `foobar` (yet))

**Reply by p3k** [reply]
Interesting, thanks for testing it! Did not expect foo and bar to be actual NPM packages – but of course they are! :sweat_smile: 

So then I probably should try to reproduce the issue with a simple GitLab setup…

**Reply by rarkins** [reply, marked as answer]
Labels are only applied at PR creation time. Maybe your PRs already existed?

**Reply by p3k** [reply]
Oh! OK gonna check that, I think I only clicked the check mark “to rebase” in the dependency dashboard…

**Reply by p3k** [reply]
huh, indeed it works as intended when there is no existing merge request around!
